### PR TITLE
Wait for tiller to be installed

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/5218

Fixes to helmclient needed for https://github.com/giantswarm/cluster-operator/pull/386. With this all chartconfig CRs and chart-operator are installed in a single reconciliation loop the first time the tenant cluster becomes available.

### CODE REVIEW

File: helmclient.go [hunk 1]
```diff
@@ -271,7 +271,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 		}
 	}
 
-	if pod != nil {
+	if !installTiller && pod != nil {
```
Comment (rossf7) [inline helmclient.go]: This fixes a bug I introduced with the tiller upgrade logic. If we are installing tiller there is no point doing the version check.

This was causing this error to be returned in some situations.

`return microerror.Maskf(executionFailedError, "invalid state cannot both install and upgrade tiller")`

File: helmclient.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 
 		o := func() error {
 			t, err := c.newTunnel()
-			if IsTillerNotFound(err) {
+			if !installTiller && IsTillerNotFound(err) {
```
Comment (rossf7) [inline helmclient.go]: This is the fix needed for https://github.com/giantswarm/cluster-operator/pull/386. Currently we always do a permanent backoff when installing tiller. This slows the bootstrap process as we need to wait for the resync period.

File: helmclient.go [hunk 3]
```diff
@@ -262,7 +262,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 
 			return nil
 		}
-		b := backoff.NewExponential(2*time.Minute, 5*time.Second)
+		b := backoff.NewExponential(1*time.Minute, 5*time.Second)
```
Comment (rossf7) [inline helmclient.go]: Current backoff is too long. I want to reduce it but not so far that it causes problems with e2e.

File: helmclient.go [hunk 4]
```diff
@@ -333,7 +333,7 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 
 			return nil
 		}
-		b := backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval)
+		b := backoff.NewExponential(1*time.Minute, 5*time.Second)
```
Comment (rossf7) [inline helmclient.go]: Again reducing the backoff.

**Comment by xh3b4sd** [comment]
Random brain fart. We think about extending the resource interfaces with the upstream [reconcile.Result](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Result). For these cases we could then say we want to re-queue after 10 seconds and not wait for the full 5 minutes. This primitive is pretty neat. Let's keep it in mind and get going with it after VOO. /cc @kopiczko @tuommaki

**Pull request merged**